File: src/main/java/nz/co/searchwellington/urls/UrlParser.java
package nz.co.searchwellington.urls;

import java.net.MalformedURLException;
import java.net.URL;

import org.springframework.stereotype.Component;

@Component
public class UrlParser {

    public String extractHostnameFrom(String fullURL) {
        try {
            final URL url = new URL(fullURL);
            return url.getHost();
            
        } catch (MalformedURLException e) {
        	return null;
        }        
    }
    
}

File: src/main/scala/nz/co/searchwellington/urls/UrlParser.scala
package nz.co.searchwellington.urls

import java.net.MalformedURLException
import java.net.URL
import org.springframework.stereotype.Component

import scala.util.Try

@Component
class UrlParser {

  def extractHostnameFrom(fullURL: String): Option[String] = Try {
    new URL(fullURL)
  }.toOption.flatMap { url =>
    Option(url.getHost)
  }
  
}
